q2_qemistree/tests/test_prune_hierarchy.py

```diff
@@ -0,0 +1,78 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from unittest import TestCase, main
+import pandas as pd
+import numpy as np
+from skbio import TreeNode
+from q2_qemistree import prune_hierarchy
+
+
+class TestPruning(TestCase):
+    def setUp(self):
+        self.tree = TreeNode.read(["((A, B)C, D)root;"])
+        self.no_column = pd.DataFrame(index=['A', 'B', 'D'],
+                                      data=[np.nan, np.nan, np.nan])
+        self.no_annotation = pd.DataFrame(index=['A', 'B', 'D'],
+                                          data=['unclassified', 'unclassified',
+                                                'unclassified'],
+                                          columns=['class'])
+        self.one_annotation = pd.DataFrame(index=['A', 'B', 'D'],
+                                           data=['foo', 'unclassified',
+                                                 'unclassified'],
+                                           columns=['class'])
+        self.no_overlap = pd.DataFrame(index=['X', 'Y', 'Z'],
+                                       data=['foo', 'bar', 'unclassified'],
+                                       columns=['class'])
+        self.one_overlap = pd.DataFrame(index=['A', 'Y', 'Z'],
+                                        data=['foo', 'bar', 'unclassified'],
+                                        columns=['class'])
+
+    def test_no_smiles(self):
+        msg = "Feature data does not contain the column 'smiles'. Molecular "
```

Comment:
The declaration of this string and all the strings in this file that are split between two lines should be enclosed in parentheses. For example, the lines:

```
msg = "Feature data does not contain the column 'smiles'. Molecular "
"hierarchy could not be pruned."
```

Should become:

```
msg = ("Feature data does not contain the column 'smiles'. Molecular "
       "hierarchy could not be pruned.")
```

Otherwise the string msg only includes the first line.